pjrt/convert/utils/xplane_visitor.zig

```diff
@@ -0,0 +1,252 @@
+const std = @import("std");
+const xplane_proto = @import("//tsl:xplane_proto");
+const xplane_schema = @import("xplane_schema.zig");
+
+pub const XEventMetadataVisitor = struct {
+    plane: *const XPlaneVisitor,
+    stats_owner: *const xplane_proto.XEventMetadata,
+
+    pub fn init(plane: *const XPlaneVisitor, metadata: *const xplane_proto.XEventMetadata) XEventMetadataVisitor {
+        return .{
+            .plane = plane,
+            .stats_owner = metadata,
+        };
+    }
+};
+
+pub const XStatVisitor = struct {
+    stat: *const xplane_proto.XStat,
+    metadata: *const xplane_proto.XStatMetadata,
+    plane: *const XPlaneVisitor,
+    type_: ?xplane_schema.StatType = null,
+
+    pub fn init(plane: *const XPlaneVisitor, stat: *const xplane_proto.XStat) XStatVisitor {
+        return XStatVisitor.internalInit(
+            plane,
+            stat,
+            plane.getStatMetadata(stat.metadata_id),
+            plane.getStatType(stat.metadata_id),
+        );
+    }
+
+    pub fn internalInit(
+        plane: *const XPlaneVisitor,
+        stat: *const xplane_proto.XStat,
+        metadata: *const xplane_proto.XStatMetadata,
+        type_: ?xplane_schema.StatType,
+    ) XStatVisitor {
+        return .{
+            .stat = stat,
+            .metadata = metadata,
+            .plane = plane,
+            .type_ = type_,
+        };
+    }
+
+    pub fn @"type"(self: *const XStatVisitor) ?xplane_schema.StatType {
+        return self.type_;
+    }
+
+    pub fn name(self: *const XStatVisitor) []const u8 {
+        return self.metadata.name.getSlice();
+    }
+
+    pub fn toString(self: *const XStatVisitor, allocator: std.mem.Allocator) ![]const u8 {
```

Comment:
I went with `AnyWriter` instead, since that works nicely with the `ArrayList`/`std.fs.File` writers.

Reply:
I think this changes havn't been pushed

Reply:
also can you rewrite this function to directly take an xplane and an xstat instead of a "xstatevisitor" ?

Since this visitor doesn't have state, it's only used to "extend" the XState proto, which is better done by just writing function using plain data as input.